File: Shared/LuceRPG.Models.Test/TestUtil.fs
﻿namespace LuceRPG.Models

module TestUtil =

    let withId (t: 'T): 'T WithId =
        let guid = System.Guid.NewGuid().ToString()

        WithId.useId guid t

    let makePlayerWithName (btmLeft: Point) (name: string): WorldObject =
        let playerData = CharacterData.randomized name
        let payload = WorldObject.create (WorldObject.Type.Player playerData) btmLeft Direction.South

        WithId.create payload

    let makePlayer (btmLeft: Point): WorldObject =
        let name = System.Guid.NewGuid().ToString()

        makePlayerWithName btmLeft name

File: Game/Assets/Tests/TestUtil.cs
using LuceRPG.Models;
using NUnit.Framework;
using System;
using UnityEngine;

public static class TestUtil
{
    public static WithId.Model<WorldObjectModule.Payload> MakePlayer(
        int x, int y, string name = null
    )
    {
        name ??= Guid.NewGuid().ToString();
        var playerData = CharacterDataModule.randomized(name);
        var topLeft = PointModule.create(x, y);

        var payload = WorldObjectModule.create(
            WorldObjectModule.TypeModule.Model.NewPlayer(playerData), topLeft, DirectionModule.Model.South);

        return WithId.create(payload);
    }
}
